pkg/issuer/acme/http/pod.go

```diff
@@ -72,6 +72,24 @@ func (s *Solver) ensurePod(ctx context.Context, ch *v1alpha1.Challenge) (*corev1
 	return s.createPod(ch)
 }
 
+// getFinalNamespace resolves the right namespace for Solver Pods for the
+// given challenge.  By default, it will be the namespace set in the
+// Challenge.  For ClusterIssuers, the it may overriden by the
+// Namespace set in the Solver's PodSpec.
+func getFinalNamespace(ch *v1alpha1.Challenge) string {
```

Comment:
I could add a check on ch.IssuerRef here to double-check that this is only allowed in ClusterIssuer, not Issuer.  Since everything is expected to go through the validation phase I didn't to keep this logic simpler, but let me know if you prefer to add it.

Reply:
Yes we shouldn't need to check here as it should be caught further up the stack, i.e. the issuer should never be marked as ready miss-configured so we assume its correct

Reply:
Agree.

Maybe using the Present function (https://github.com/rikatz/cert-manager/blob/4a31bd25ff5cad3910a91cf8cf93ae7d2fa165b5/pkg/issuer/acme/http/http.go#L112) is better, and you cover the whole cases of Pod, Service and Ingress

Reply:
:+1: thanks @rikatz good point, I'll do that.